velox/functions/lib/LikeFunctionUtils.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/lib/LikeFunctionUtils.h"
+#include "velox/vector/TypeAliases.h"
+
+namespace facebook::velox::functions {
+namespace likeutils {
+
+bool isWildcardInPattern(const std::string& pattern) {
+  vector_size_t patternLength = pattern.size();
+  for (vector_size_t i = 0; i < patternLength; i++) {
+    if ((pattern[i] == '%') || (pattern[i] == '_')) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool matchExactPattern(const std::string& input, const std::string& pattern) {
+  vector_size_t i = 0;
+  vector_size_t j = 0;
+  vector_size_t inputLength = input.size();
+  vector_size_t patternLength = pattern.size();
+
+  if (inputLength != patternLength) {
+    return false;
+  }
+
+  for (; (i < inputLength) && (j < patternLength); i++, j++) {
```

Comment:
Since inputLength == patternLength, we don't need two variable i and j. A single variable would work. Also, it would be simpler to just use memcmp function.

```
for (auto i = 0; i < inputLength; ++i) {
   if (input[i] != pattern[i]) {
      return false;
   }
}

return true;
```